### DISCUSSION TITLE
poll: <For> with custom equality/keying

### DISCUSSION THREAD

**Discussion by LiQuidProQuo** [solidjs/solid#1523] [Polls]
`<For>` component only supports referential equality.
do we want more?

## key_by
```js
// provide a transformer to generate a unique key
<For  each={[ {id:1 value:0}, {id:3 value}  ]} 
         key_by{(o)=>o.id}
>
</For>
```

## equals 
```js
// provide the equality function 
<For equals={(a,b)=>boolan}>
</For>
```

## composable `key_by` and `equals`
```js
<For  each={[ {id:1 value:0},  {id:3 value}  ]} 
         key_by={(o)=>o.id} 
         equals={ (a,b)=>  a === b  }  // a and b are transformed by key_by callback
  >
</For>
```

## 3rd party primitive 
[solidjs-community/&lt;Key>](https://github.com/solidjs-community/solid-primitives/tree/main/packages/keyed#key)
```jsx
<Key each={items()} by={item => item.id} fallback={<div>No items</div>}>
  {item => <div>{item()}</div>}
</Key>
```

**Comment by lxsmnsyc** [comment]
I think this is just a repeat of https://github.com/solidjs/solid/discussions/813 which motivated the creation of `solid-primitives/keyed`
